notus/scanner/connect/ssh.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (C) 2020-2021 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from typing import Optional, Tuple
+import re
+
+from paramiko import AuthenticationException, SSHClient, AutoAddPolicy
+
+
+logger = logging.getLogger(__name__)
+
+# For debian releases
+_debian_re = re.compile(r"^([0-9]+)([0-9.]+)")
+# For EuerOS 2.0 releases
+_euler_re = re.compile(
```

Comment:
I think it would be better to create specifiec classes, closures for each distribution instead of doing that all directly in ssh.py; that way we could more easily extend it with other distributions without having to change the actual ssh implementation every time.